test/template/test_template.py

```diff
@@ -589,3 +589,64 @@ def test_template_split_multicell():
     tmpl = Template(format="A4", unit="pt", elements=elements)
     res = tmpl.split_multicell(text, "multline_text")
     assert res == expected
+
+
+def test_template_wrapmode(tmp_path):
+    # Test that wrap mode can optionally be used to set wrapping using characters instead of words
+    elements = [
+        {
+            "name": "multi",
+            "type": "T",
+            "x1": 80,
+            "y1": 10,
+            "x2": 170,
+            "y2": 15,
+            "text": "If multiline is False, the text should still not wrap even if wrapmode is specified.",
+            "background": 0xEEFFFF,
+            "multiline": False,
+            "wrapmode": "WORD",
+        },
+        {
+            "name": "multi",
+            "type": "T",
+            "x1": 80,
+            "y1": 40,
+            "x2": 170,
+            "y2": 45,
+            "text": "If multiline is True, and wrapmode is omitted, the text should wrap by word and not "
+            + "cause an error due to omission of the wrapmode argument.",
+            "background": 0xEEFFFF,
+            "multiline": True,
+        },
+        {
+            "name": "multi",
+            "type": "T",
+            "x1": 80,
+            "y1": 70,
+            "x2": 170,
+            "y2": 75,
+            "text": "If multiline is True and the wrapmode argument is provided, it should not cause a "
+            + "problem even if using the default wrapmode of 'WORD'.",
+            "background": 0xEEFFFF,
+            "multiline": True,
+            "wrapmode": "WORD",
+        },
+        {
+            "name": "multi",
+            "type": "T",
+            "x1": 80,
+            "y1": 100,
+            "x2": 170,
+            "y2": 105,
+            "text": "If multiline is True and the wrapmode argument is provided, it should result in "
```

Comment:
Minor nitpick: instead of
`   the wrapmode argument is provided`
this whould read
`   the wrapmode is "CHAR"`
or it might confuse some future reader.

While this is technically test data, it also serves as a code comment here, and those should never contradict the actual code.

Reply:
Thanks @gmischler I've just made this update!